### PR TITLE
fix: storage migration permissions

### PR DESCRIPTION
**Pull request by sachscode**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts [hunk 1]
```diff
@@ -513,3 +546,9 @@ export class S3InputState {
     }
   }
 }
+
+//Helper
+function removeDuplicates(strArray : Array<string>) : string[] {
+  return strArray.splice(0, strArray.length, ...(new Set(strArray)))
+};
+
```
Comment (jhockett) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:554]: nit: Can use `_.uniq()`

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts [hunk 2]
```diff
@@ -148,6 +148,33 @@ export class S3InputState {
     return oldParams;
   }
 
+  inferAuthPermissions(oldParams: $TSAny): any[] {
```
Comment (jhockett) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:151]: nit: use $TSAny consistently

File: packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts [hunk 3]
```diff
@@ -390,15 +422,16 @@ export class S3InputState {
   }
 
   public static getInputPermissionsFromCfnPermissions(selectedGuestPermissions: S3CFNPermissionType[] | undefined) {
-    if (selectedGuestPermissions) {
-      return selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+    if (selectedGuestPermissions && selectedGuestPermissions.length > 0) {
+      const inputParams = selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+      return removeDuplicates(inputParams) as Array<S3PermissionType>; //required to remove List and Read mapping to the same entity
```
Comment (kaustavghosh06) [inline packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:427]: just convert to Set() and then re-convert to List. I think it's a bit ore readable than the function below.

Comment (kaustavghosh06) [reply packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts:427]: or use loadash

**Review by jhockett — APPROVED** [review]

File: packages/amplify-e2e-core/src/categories/storage.ts [hunk 4]
```diff
@@ -539,8 +539,6 @@ export function updateS3AddTriggerWithAuthOnlyReqMigration(cwd: string, settings
       .sendCarriageReturn() // Auth only users
       .wait('What kind of access do you want for Authenticated users?')
       .sendCarriageReturn() // Select preselected permissions
-      .wait('What kind of access do you want for Guest users?')
```
Comment (jhockett) [inline packages/amplify-e2e-core/src/categories/storage.ts:542]: why is this getting removed?

**Pull request merged**